services/gateway/endpoints/src/main/java/org/eclipse/ditto/services/gateway/endpoints/directives/ContentTypeValidationDirective.java

```diff
@@ -0,0 +1,111 @@
+      * @param ctx the context of the request.
+      * @param dittoHeaders the ditto-headers of a request.
+      * @param inner route to wrap.
+      * @return the wrapped route.
+      */
+     public static Route ensureValidContentType(final Set<String> supportedMediaTypes, final RequestContext ctx,
+             final DittoHeaders dittoHeaders,
+             final Supplier<Route> inner) {
+         return enhanceLogWithCorrelationId(dittoHeaders.getCorrelationId(), () -> {
+             final String requestsMediaType = extractMediaType(ctx.getRequest());
+
+             if(supportedMediaTypes.contains(requestsMediaType)){
+                 return inner.get();
+             } else {
+                 LOGGER.info("Request rejected: unsupported media-type: <{}>  request: <{}>",
+                         requestsMediaType, requestToLogString(ctx.getRequest()));
+                 return completeWithMediaTypeNotSupported(requestsMediaType, supportedMediaTypes);
+             }
+         });
+     }
+
+     protected static Route completeWithMediaTypeNotSupported(final String mediaType,
+             final Set<String> supportedByResource){
+
+             final String msgPatten = "The Media-Type <{0}> is not supported for this endpoint. Allowed " +
+                     "Media-Types are: <{1}>";
+             final String responseMessage =
+                     MessageFormat.format(msgPatten, mediaType, supportedByResource);
+
+             return complete(StatusCodes.UNSUPPORTED_MEDIA_TYPE, responseMessage);
+     }
+
+     protected static String requestToLogString(final HttpRequest request){
+         final String msgPatten = "{0} {1} {2}";
+         return MessageFormat.format(msgPatten,
+                 request.getUri().getHost().address(),
+                 request.method().value(),
+                 request.getUri().getPathString());
+     }
+
+
+     /**
+      * Uses either the raw-header or the content-type parsed by akka-http.
+      * The parsed content-type is never null, because akka-http sets a default.
+      * In the case of akka's default value, the raw version is preferred.
+      * The raw content-type header is not available, in case akka successfully parsed the content-type.
+      * For akka-defaults:
+      * {@link akka.http.impl.engine.parsing.HttpRequestParser#createLogic} <code>parseEntity</code>
+      * and {@link akka.http.scaladsl.model.HttpEntity$}.
+      * @see
+      * <a href="https://doc.akka.io/docs/akka-http/current/common/http-model.html#http-headers">Akkas Header model</a>
+      *
+      * @param request the request where the media type will be extracted from.
+      * @return the extracted media-type.
+      */
+     protected static String extractMediaType(HttpRequest request){
+         final Optional<HttpHeader> rawContentType = StreamSupport.stream(request.getHeaders().spliterator(), false)
+                 .filter(header -> header.name().equals(DittoHeaderDefinition.CONTENT_TYPE.getKey()))
```

Comment:
hm, here you use the `rawContentType` as preferred one which could still contain a charset, doesn't it?

Reply:
yes it does. But unfortunately akka's header handling is not quite obvious. If the content-type can be parsed by akka, then there is no 'content-type' raw-header. It tried to document it as good as i can and also linked to the akka documentation.